src/shared/sol-conffile.c

```diff
@@ -917,9 +943,162 @@ _load_vector_defaults(void)
 }
 
 static int
+iterate_dir_cb(void *data, const char *dir_path, struct dirent *en)
+{
+    char path[PATH_MAX], *sep, *endptr = NULL, *str;
+    struct sol_ptr_vector aliases = SOL_PTR_VECTOR_INIT;
+    struct sol_buffer *file_contents = data;
+    struct sol_json_scanner scanner;
+    struct sol_json_token value;
+    enum sol_json_loop_reason end_reason;
+    const size_t extension = strlen(".json");
```

Comment:
Why don't you use `sizeof()` ? `sizeof` count the size in compile time while `strlen` is in run time. Maybe the compilers are smart enough to do this in runtime, but I think it's better do not presume it.